receiver/awsxrayreceiver/internal/translator/sql.go

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator
+
+import (
+	"fmt"
+	"regexp"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/awsxray"
+)
+
+func addSQLToSpan(sql *awsxray.SQLData, attrs *pdata.AttributeMap) error {
+	if sql == nil {
+		return nil
+	}
+
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c615d2db351929b99e46f7b427f39c12afe15b54/exporter/awsxrayexporter/translator/sql.go#L60
+	if sql.URL != nil {
+		dbURL, dbName, err := splitSQLURL(*sql.URL)
+		if err != nil {
+			return err
+		}
+		attrs.UpsertString(conventions.AttributeDBConnectionString, dbURL)
+		attrs.UpsertString(conventions.AttributeDBName, dbName)
+	}
+	// not handling sql.ConnectionString for now because the X-Ray exporter
+	// does not support it
+	addString(sql.DatabaseType, conventions.AttributeDBSystem, attrs)
+	addString(sql.SanitizedQuery, conventions.AttributeDBStatement, attrs)
+	addString(sql.User, conventions.AttributeDBUser, attrs)
+	return nil
+}
+
+// SQL URL is of the format: protocol+transport://host:port/dbName
```

Comment:
Thanks for the comment - it seems like the regex won't handle query parameters in th eURL like many JDBC urls? `mysql://localhost:1234/mydb?myInterceptor=foo`